fix(autodev): resolve duplicate HistoryRepository impl causing compilation failure

## Summary

현재 uncommitted 변경에서 `repository.rs`에 `HistoryRepository` 관련 코드가 **중복 정의**되어 컴파일이 실패합니다.

### Compilation Errors
```
error[E0428]: the name `map_history_row` is defined multiple times
    --> src/infra/db/repository.rs:2037:1
error[E0428]: the name `tests` is defined multiple times
    --> src/infra/db/repository.rs:2058:1
error[E0119]: conflicting implementations of trait `core::repository::HistoryRepository` for type `Database`
    --> src/infra/db/repository.rs:1583:1
```

### Duplicate Definitions
| 항목 | 1차 정의 | 2차 정의 (중복) |
|------|---------|----------------|
| `impl HistoryRepository for Database` | line 1514 | line 1583 |
| `fn map_history_row` | line 1846 | line 2037 |
| `mod tests` | line 1867 | line 2058 |

### Affected Files (uncommitted)
- `cli/src/core/models.rs`
- `cli/src/core/repository.rs`
- `cli/src/infra/db/mod.rs`
- `cli/src/infra/db/repository.rs` ← 컴파일 에러 원인
- `cli/src/infra/db/schema.rs`
- `cli/src/service/daemon/mod.rs`

### Root Cause
v5 `HistoryRepository` (append-only transition history) 구현 중 기존 impl 블록이 남아있는 상태에서 새 impl이 추가됨.

### Fix
중복된 두 번째 블록(line 1583, 2037, 2058)을 제거하거나, 두 블록의 차이를 머지하여 하나로 통합.

### Acceptance Criteria
- [ ] `cargo test` 통과
- [ ] `cargo clippy -- -D warnings` 통과
- [ ] HistoryRepository impl이 하나만 존재

--- a/plugins/autodev/cli/src/core/config/models.rs
+++ b/plugins/autodev/cli/src/core/config/models.rs
@@ -908,7 +908,7 @@ sources:
         trigger:
           label: "custom:implement"
 "#;
-        let cfg: WorkflowConfig = serde_yaml::from_str(yaml).unwrap();
+        let cfg: WorkflowConfig = serde_yml::from_str(yaml).unwrap();
         assert_eq!(
             cfg.sources.github.trigger_label("analyze"),
             Some("custom:analyze")
@@ -937,7 +937,7 @@ sources:
       analyze:
         trigger: {}
 "#;
-        let cfg: WorkflowConfig = serde_yaml::from_str(yaml).unwrap();
+        let cfg: WorkflowConfig = serde_yml::from_str(yaml).unwrap();
         // trigger exists but label is None
         assert_eq!(cfg.sources.github.trigger_label("analyze"), None);
     }

--- a/plugins/autodev/cli/src/core/repository.rs
+++ b/plugins/autodev/cli/src/core/repository.rs
@@ -135,6 +135,7 @@ pub trait HistoryRepository {
         limit: usize,
     ) -> Result<Vec<HistoryEntry>>;
 }
+
 pub trait CronRepository {
     fn cron_add(&self, job: &NewCronJob) -> Result<String>;
     fn cron_list(&self, repo: Option<&str>) -> Result<Vec<CronJob>>;

--- a/plugins/autodev/cli/src/infra/db/repository.rs
+++ b/plugins/autodev/cli/src/infra/db/repository.rs
@@ -1588,6 +1588,75 @@ impl CronRepository for Database {
     }
 }
 
+impl HistoryRepository for Database {
+    fn history_insert(&self, entry: &NewHistoryEntry) -> Result<String> {
+        let conn = self.conn();
+        let id = Uuid::new_v4().to_string();
+        let now = Utc::now().to_rfc3339();
+
+        conn.execute(
+            "INSERT INTO history (id, source_id, workspace_id, task_kind, status, error_message, duration_ms, created_at) \
+             VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8)",
+            rusqlite::params![
+                id,
+                entry.source_id,
+                entry.workspace_id,
+                entry.task_kind,
+                entry.status.as_str(),
+                entry.error_message,
+                entry.duration_ms,
+                now,
+            ],
+        )?;
+        Ok(id)
+    }
+
+    fn history_count_failures(&self, source_id: &str) -> Result<i64> {
+        let conn = self.conn();
+        let count: i64 = conn.query_row(
+            "SELECT COUNT(*) FROM history WHERE source_id = ?1 AND status = 'failed'",
+            rusqlite::params![source_id],
+            |row| row.get(0),
+        )?;
+        Ok(count)
+    }
+
+    fn history_list_by_source(&self, source_id: &str, limit: usize) -> Result<Vec<HistoryEntry>> {
+        let conn = self.conn();
+        let mut stmt = conn.prepare(
+            "SELECT id, source_id, workspace_id, task_kind, status, error_message, duration_ms, created_at \
+             FROM history WHERE source_id = ?1 ORDER BY created_at DESC LIMIT ?2",
+        )?;
+        let rows = stmt.query_map(rusqlite::params![source_id, limit as i64], map_history_row)?;
+        let mut entries = Vec::new();
+        for row in rows {
+            entries.push(row?);
+        }
+        Ok(entries)
+    }
+
+    fn history_list_by_workspace(
+        &self,
+        workspace_id: &str,
+        limit: usize,
+    ) -> Result<Vec<HistoryEntry>> {
+        let conn = self.conn();
+        let mut stmt = conn.prepare(
+            "SELECT id, source_id, workspace_id, task_kind, status, error_message, duration_ms, created_at \
+             FROM history WHERE workspace_id = ?1 ORDER BY created_at DESC LIMIT ?2",
+        )?;
+        let rows = stmt.query_map(
+            rusqlite::params![workspace_id, limit as i64],
+            map_history_row,
+        )?;
+        let mut entries = Vec::new();
+        for row in rows {
+            entries.push(row?);
+        }
+        Ok(entries)
+    }
+}
+
 // ─── Row-mapping helpers ───
 
 /// Converts a `query_row` result into `Ok(Some(x))` / `Ok(None)` / `Err(e)`,
@@ -1777,78 +1846,6 @@ fn map_cron_row(row: &rusqlite::Row<'_>) -> Result<CronJob> {
         last_run_at: row.get(8)?,
         created_at: row.get(9)?,
     })
-}
-
-impl HistoryRepository for Database {
-    fn history_insert(&self, entry: &NewHistoryEntry) -> Result<String> {
-        let conn = self.conn();
-        let id = Uuid::new_v4().to_string();
-        let now = Utc::now().to_rfc3339();
-
-        conn.execute(
-            "INSERT INTO history \
-             (id, source_id, workspace_id, task_kind, status, error_message, duration_ms, created_at) \
-             VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8)",
-            rusqlite::params![
-                id,
-                entry.source_id,
-                entry.workspace_id,
-                entry.task_kind,
-                entry.status.as_str(),
-                entry.error_message,
-                entry.duration_ms,
-                now,
-            ],
-        )?;
-        Ok(id)
-    }
-
-    fn history_count_failures(&self, source_id: &str) -> Result<i64> {
-        let conn = self.conn();
-        let count: i64 = conn.query_row(
-            "SELECT COUNT(*) FROM history WHERE source_id = ?1 AND status = 'failed'",
-            rusqlite::params![source_id],
-            |row| row.get(0),
-        )?;
-        Ok(count)
-    }
-
-    fn history_list_by_source(&self, source_id: &str, limit: usize) -> Result<Vec<HistoryEntry>> {
-        let conn = self.conn();
-        let mut stmt = conn.prepare(
-            "SELECT id, source_id, workspace_id, task_kind, status, \
-             error_message, duration_ms, created_at \
-             FROM history WHERE source_id = ?1 ORDER BY created_at DESC LIMIT ?2",
-        )?;
-        let rows = stmt.query_map(rusqlite::params![source_id, limit as i64], map_history_row)?;
-        let mut entries = Vec::new();
-        for row in rows {
-            entries.push(row?);
-        }
-        Ok(entries)
-    }
-
-    fn history_list_by_workspace(
-        &self,
-        workspace_id: &str,
-        limit: usize,
-    ) -> Result<Vec<HistoryEntry>> {
-        let conn = self.conn();
-        let mut stmt = conn.prepare(
-            "SELECT id, source_id, workspace_id, task_kind, status, \
-             error_message, duration_ms, created_at \
-             FROM history WHERE workspace_id = ?1 ORDER BY created_at DESC LIMIT ?2",
-        )?;
-        let rows = stmt.query_map(
-            rusqlite::params![workspace_id, limit as i64],
-            map_history_row,
-        )?;
-        let mut entries = Vec::new();
-        for row in rows {
-            entries.push(row?);
-        }
-        Ok(entries)
-    }
 }
 
 fn map_history_row(row: &rusqlite::Row<'_>) -> rusqlite::Result<HistoryEntry> {

--- a/plugins/autodev/cli/src/infra/db/schema.rs
+++ b/plugins/autodev/cli/src/infra/db/schema.rs
@@ -238,6 +238,19 @@ pub fn create_tables(conn: &Connection) -> Result<()> {
         );
         CREATE INDEX IF NOT EXISTS idx_feedback_patterns_repo ON feedback_patterns(repo_id, status);
 
+        CREATE TABLE IF NOT EXISTS history (
+            id              TEXT PRIMARY KEY,
+            source_id       TEXT NOT NULL,
+            workspace_id    TEXT NOT NULL,
+            task_kind       TEXT NOT NULL,
+            status          TEXT NOT NULL,
+            error_message   TEXT,
+            duration_ms     INTEGER,
+            created_at      TEXT NOT NULL
+        );
+        CREATE INDEX IF NOT EXISTS idx_history_source ON history(source_id, created_at);
+        CREATE INDEX IF NOT EXISTS idx_history_workspace ON history(workspace_id, created_at);
+
         COMMIT;
         ",
     )?;

